### PR TITLE
Rework meta data

### PR DESCRIPTION
**Pull request by monospaced**
> [!IMPORTANT]  
> On merge I need to update production all Page `title` and `description` fields in the Puck editor. 
> And add `authorUrl` to the two blog posts (Chris' LinkedIn).

* Add dynamic page-specific JSON-LD schema
* Add dynamic page-specific canonical URLs
* Add dynamic page-specific Open Graph / Twitter metadata (though still a single hardcoded image)
* Add `"@type": "WebSite"` schema to homepage to [provide a site name to Google Search](https://developers.google.com/search/docs/appearance/site-names)
* Fix duplicate `<meta name="viewport" />` bug
* Update everything (that makes sense) to use [Next.js Metadata API](https://nextjs.org/docs/app/building-your-application/optimizing/metadata) 
* Appropriate `<title>` suffixes (e.g. ` | Measured`) now added automatically
* Centralise global content strings in `/content.json`

### Home

```html
<!DOCTYPE html>
<html lang="en-gb" style="scroll-behavior: smooth">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, viewport-fit=cover"
    />
    …
    <meta name="theme-color" content="#a81a66" />
    <title>Measured</title>
    <meta
      name="description"
      content="The React and UI specialists. Measured bring decades of experience in web development, design systems and React to solve your toughest UI challenges."
    />
    <link rel="manifest" href="/manifest.webmanifest" />
    <link rel="canonical" href="https://measured.co" />
    <meta property="og:title" content="Measured" />
    <meta
      property="og:description"
      content="The React and UI specialists. Measured bring decades of experience in web development, design systems and React to solve your toughest UI challenges."
    />
    <meta property="og:url" content="https://measured.co" />
    <meta property="og:site_name" content="Measured" />
    <meta property="og:locale" content="en_GB" />
    <meta property="og:image:alt" content="Measured" />
    <meta property="og:image:height" content="600" />
    <meta property="og:image" content="https://measured.co/social.png" />
    <meta property="og:image:type" content="image/png" />
    <meta property="og:image:width" content="1200" />
    <meta property="og:type" content="website" />
    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:site" content="@hellomeasuredco" />
    <meta name="twitter:title" content="Measured" />
    <meta
      name="twitter:description"
      content="The React and UI specialists. Measured bring decades of experience in web development, design systems and React to solve your toughest UI challenges."
    />
    <meta name="twitter:image:alt" content="Measured" />
    <meta name="twitter:image:height" content="600" />
    <meta name="twitter:image" content="https://measured.co/social.png" />
    <meta name="twitter:image:type" content="image/png" />
    <meta name="twitter:image:width" content="1200" />
    <meta
      http-equiv="delegate-ch"
      content="sec-ch-width https://res.cloudinary.com; sec-ch-dpr https://res.cloudinary.com; sec-ch-viewport-width https://res.cloudinary.com;"
    />
    <link
      rel="alternate"
      type="application/atom+xml"
      href="/feed.atom"
      title="Measured - Atom Feed"
    />
    <link
      rel="alternate"
      type="application/rss+xml"
      href="/feed.xml"
      title="Measured - RSS Feed"
    />
    <link
      rel="alternate"
      type="application/json"
      href="/feed.json"
      title="Measured - JSON Feed"
    />
    <link rel="icon" href="/favicon.ico" sizes="48x48" />
    <link rel="icon" href="/favicon.svg" type="image/svg+xml" />
    <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
    <script
      defer=""
      data-domain="measured.co"
      src="https://plausible.io/js/plausible.js"
    ></script>
    …
  </head>
  <body style="margin: 0">
    <script type="application/ld+json">
      [
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "logo": "https://measured.co/logo-schema-organization.svg",
          "name": "Measured",
          "sameAs": [
            "https://github.com/measuredco",
            "https://twitter.com/hellomeasuredco",
            "https://www.linkedin.com/company/measuredco/"
          ],
          "url": "https://measured.co"
        },
        {
          "@context": "https://schema.org",
          "@type": "WebSite",
          "name": "Measured",
          "url": "https://measured.co"
        }
      ]
    </script>
    <div class="msrd">
      …
```

### Blog

```html
<!DOCTYPE html>
<html lang="en-gb" style="scroll-behavior: smooth">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, viewport-fit=cover"
    />
    …
    <meta name="theme-color" content="#a81a66" />
    <title>Blog | Measured</title>
    <meta
      name="description"
      content="A blog from Measured, the React and UI specialists."
    />
    <link rel="manifest" href="/manifest.webmanifest" />
    <link rel="canonical" href="https://measured.co/blog" />
    <meta property="og:title" content="Blog" />
    <meta
      property="og:description"
      content="A blog from Measured, the React and UI specialists."
    />
    <meta property="og:url" content="https://measured.co/blog" />
    <meta property="og:site_name" content="Measured" />
    <meta property="og:locale" content="en_GB" />
    <meta property="og:image:alt" content="Measured" />
    <meta property="og:image:height" content="600" />
    <meta property="og:image" content="https://measured.co/social.png" />
    <meta property="og:image:type" content="image/png" />
    <meta property="og:image:width" content="1200" />
    <meta property="og:type" content="website" />
    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:title" content="Blog" />
    <meta
      name="twitter:description"
      content="A blog from Measured, the React and UI specialists."
    />
    <meta name="twitter:image:alt" content="Measured" />
    <meta name="twitter:image:height" content="600" />
    <meta name="twitter:image" content="https://measured.co/social.png" />
    <meta name="twitter:image:type" content="image/png" />
    <meta name="twitter:image:width" content="1200" />
    <meta
      http-equiv="delegate-ch"
      content="sec-ch-width https://res.cloudinary.com; sec-ch-dpr https://res.cloudinary.com; sec-ch-viewport-width https://res.cloudinary.com;"
    />
    <link
      rel="alternate"
      type="application/atom+xml"
      href="/feed.atom"
      title="Measured - Atom Feed"
    />
    <link
      rel="alternate"
      type="application/rss+xml"
      href="/feed.xml"
      title="Measured - RSS Feed"
    />
    <link
      rel="alternate"
      type="application/json"
      href="/feed.json"
      title="Measured - JSON Feed"
    />
    <link rel="icon" href="/favicon.ico" sizes="48x48" />
    <link rel="icon" href="/favicon.svg" type="image/svg+xml" />
    <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
    <script
      defer=""
      data-domain="measured.co"
      src="https://plausible.io/js/plausible.js"
    ></script>
    …
  </head>
  <body style="margin: 0">
    <script type="application/ld+json">
      {
        "@context": "https://schema.org",
        "@type": "ItemList",
        "itemListElement": [
          {
            "@type": "ListItem",
            "position": 1,
            "url": "https://measured.co/blog/building-a-react-page-builder-an-introduction-to-puck"
          },
          {
            "@type": "ListItem",
            "position": 2,
            "url": "https://measured.co/blog/puck-open-source-visual-react-editor-modern-content-workflows"
          }
        ]
      }
    </script>
    <div class="msrd">
      …
```

### Post

```html
<!DOCTYPE html>
<html lang="en-gb" style="scroll-behavior: smooth">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, viewport-fit=cover"
    />
    …
    <meta name="theme-color" content="#a81a66" />
    <title>
      Building a React Page Builder: An Introduction to Puck | Blog | Measured
    </title>
    <meta name="description" content="A blog post from Measured." />
    <link rel="manifest" href="/manifest.webmanifest" />
    <link
      rel="canonical"
      href="https://measured.co/blog/building-a-react-page-builder-an-introduction-to-puck"
    />
    <meta
      property="og:title"
      content="Building a React Page Builder: An Introduction to Puck"
    />
    <meta property="og:description" content="A blog post from Measured." />
    <meta
      property="og:url"
      content="https://measured.co/blog/building-a-react-page-builder-an-introduction-to-puck"
    />
    <meta property="og:site_name" content="Measured" />
    <meta property="og:locale" content="en_GB" />
    <meta property="og:image:alt" content="Measured" />
    <meta property="og:image:height" content="600" />
    <meta property="og:image" content="https://measured.co/social.png" />
    <meta property="og:image:type" content="image/png" />
    <meta property="og:image:width" content="1200" />
    <meta property="og:type" content="article" />
    <meta
      property="article:published_time"
      content="2023-09-28T00:00:00.000Z"
    />
    <meta property="article:modified_time" content="2023-09-28T00:00:00.000Z" />
    <meta property="article:author" content="Chris Villa" />
    <meta name="twitter:card" content="summary_large_image" />
    <meta
      name="twitter:title"
      content="Building a React Page Builder: An Introduction to Puck"
    />
    <meta name="twitter:description" content="A blog post from Measured." />
    <meta name="twitter:image:alt" content="Measured" />
    <meta name="twitter:image:height" content="600" />
    <meta name="twitter:image" content="https://measured.co/social.png" />
    <meta name="twitter:image:type" content="image/png" />
    <meta name="twitter:image:width" content="1200" />
    <meta
      http-equiv="delegate-ch"
      content="sec-ch-width https://res.cloudinary.com; sec-ch-dpr https://res.cloudinary.com; sec-ch-viewport-width https://res.cloudinary.com;"
    />
    <link
      rel="alternate"
      type="application/atom+xml"
      href="/feed.atom"
      title="Measured - Atom Feed"
    />
    <link
      rel="alternate"
      type="application/rss+xml"
      href="/feed.xml"
      title="Measured - RSS Feed"
    />
    <link
      rel="alternate"
      type="application/json"
      href="/feed.json"
      title="Measured - JSON Feed"
    />
    <link rel="icon" href="/favicon.ico" sizes="48x48" />
    <link rel="icon" href="/favicon.svg" type="image/svg+xml" />
    <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
    <script
      defer=""
      data-domain="measured.co"
      src="https://plausible.io/js/plausible.js"
    ></script>
    …
  </head>
  <body style="margin: 0">
    <script type="application/ld+json">
      {
        "@context": "https://schema.org",
        "@type": "BlogPosting",
        "author": {
          "@type": "Person",
          "name": "Chris Villa",
          "url": "https://measured.co"
        },
        "dateModified": "2023-09-28T00:00:00.000Z",
        "datePublished": "2023-09-28T00:00:00.000Z",
        "headline": "Building a React Page Builder: An Introduction to Puck",
        "publisher": {
          "@type": "Organization",
          "name": "Measured",
          "url": "https://measured.co"
        }
      }
    </script>
    <div class="msrd">
      …
```

### CODE REVIEW

**Review by anglepoised — APPROVED** [review]
🚀

File: app/page.tsx [hunk 1]
```diff
@@ -1 +1,96 @@
...
+    },
+  ];
+
+  return (
+    <>
+      <script
+        dangerouslySetInnerHTML={{ __html: `${JSON.stringify(schema)}` }}
+        type="application/ld+json"
+      />
+      {}
```
Comment (chrisvxd) [inline app/page.tsx:92]: Miscellaneous `{}`?

File: app/blog/page.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+const getPageRes = (path: string) =>
+  supabase.from("puck").select("*").eq("path", path).maybeSingle();
+
+export { viewport } from "../page";
+
+export async function generateMetadata(): Promise<Metadata> {
+  const path = "/blog";
+  const pageRes = await getPageRes(path);
+  const data = pageRes?.data?.data as Data;
```
Comment (chrisvxd) [inline app/blog/page.tsx:21]: Seems like a future opportunity for a refactor as it's now used in a few places

```tsx
  const { data } = await getPageData("/blog");
```

File: app/blog/[...puckPath]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+export const dynamic = "force-static";
+
+export async function generateMetadata({
+  params,
+}: {
+  params: { puckPath: string[] };
+}): Promise<Metadata> {
+  const blogPath = "/blog";
+  const { path } = resolvePuckPath(params.puckPath);
```
Comment (chrisvxd) [inline app/blog/[...puckPath]/page.tsx:26]: We might be able to `resolvePuckPath` soon because we have a distinct edit route

**Review by chrisvxd — COMMENTED** [review]
Looks great! Few non-blocking comments

File: app/page.tsx (hunk 1 above)
Comment (monospaced) [reply app/page.tsx:92]: Good spot ty!

File: app/blog/page.tsx (hunk 2 above)
Comment (monospaced) [reply app/blog/page.tsx:21]: Fair

**Pull request merged**